contracts/delegatecall/LibEtherForENS.sol

```diff
@@ -0,0 +1,25 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.20;
+
+abstract contract Resolver {
+    function addr(bytes32 node) public view virtual returns (address);
+}
+
+abstract contract ENS {
+    function resolver(bytes32 node) public view virtual returns (Resolver);
+}
+
+library LibEtherForENS {
+    function etherForENS(bytes32 node, uint256 value, bytes memory payload) external{
+        // address of The ENS registry
+        ENS ens = ENS(0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e);
```

Comment:
This address is the same for the networks that ENS root registry has been deployed [so far](https://docs.ens.domains/learn/deployments). It will likely be the same for other future networks as well. But if that's not the case, then this address should be modify accordingly to deploy this library onto that network.

Reply:
I'd be more comfortable with this address being provided in the constructor. This is, considering we convert this library into a contract. Then, we can provide the correct address on the deployment script depending on the chain.

Reply:
One nice experiment would be to see if the contract can access `immutable` variables such as the ENS registry contract even when it's called via `delegatecall`. My intuition tells me that immutable variables are written to byte code at contract creation time.

Reply:
Yeah. `immutable` variables should be in the bytecode during deployment. I have done an experiment to confirm that ✅